app/scripts/modules/core/src/managed/overview/artifact/utils.ts

```diff
@@ -0,0 +1,128 @@
+import { DateTime } from 'luxon';
+
+import { ManagedWriter, showModal, useApplicationContext } from 'core/index';
+import { timeDiffToString } from 'core/utils';
+
+import { MarkAsBadActionModal, PinActionModal, UnpinActionModal } from './ActionModal';
+import { VersionAction } from './VersionMetadata';
+import { useFetchApplicationLazyQuery } from '../../graphql/graphql-sdk';
+import { QueryArtifactVersion, QueryLifecycleStep } from '../types';
+import { DEFAULT_VERSION_STATUSES } from '../utils';
+
+export const getLifecycleEventDuration = (
+  version: QueryArtifactVersion | undefined,
+  type: QueryLifecycleStep['type'],
+) => {
+  const event = version?.lifecycleSteps?.find((step) => step.type === type);
+  if (!event) return undefined;
+  const { startedAt, completedAt } = event;
+  if (startedAt && completedAt) {
+    return timeDiffToString(DateTime.fromISO(startedAt), DateTime.fromISO(completedAt));
+  }
+  return undefined;
+};
+
+export const getLifecycleEventLink = (version: QueryArtifactVersion | undefined, type: QueryLifecycleStep['type']) => {
+  return version?.lifecycleSteps?.find((step) => step.type === type)?.link;
+};
+
+const MODAL_MAX_WIDTH = 750;
+
+export const useCreateVersionActions = ({
+  environment,
+  reference,
+  version,
+  buildNumber,
+  commitMessage,
+  isPinned,
+}: {
+  environment: string;
+  reference: string;
+  version: string;
+  buildNumber?: string;
+  commitMessage?: string;
+  isPinned: boolean;
+}): VersionAction[] | undefined => {
+  const application = useApplicationContext();
+  if (!application) throw new Error('Application context is empty');
+  const [refetch] = useFetchApplicationLazyQuery({
+    variables: { appName: application.name, statuses: DEFAULT_VERSION_STATUSES },
+    fetchPolicy: 'network-only',
+  });
+  const actions: VersionAction[] = [
```

Comment:
this section is a bit difficult to parse, perhaps it would improve readability to extract the onClick handlers as variables?

Reply:
What do you mean as variables?

Reply:
nvm, moved them